### ISSUE TITLE
Multipolygon.clone() is incomplete: Modifying the clone affects the original

### ISSUE THREAD

**Issue by awaterme** [openlayers/ol3#3326]
Multipolygon.clone() currently doesnt clone the `endss` field, which is a **nested** array.

Consequence: Clones are not completly decoupled, but changing the clone also modifies the original.

Please perform a "deep clone" of the `endss` fields, rather than a flat slice() on the `endss`-array.

Current implementation:

```javascript
ol.geom.MultiPolygon.prototype.clone = function() {
  var multiPolygon = new ol.geom.MultiPolygon(null);
  multiPolygon.setFlatCoordinates(
      this.layout, this.flatCoordinates.slice(), this.endss_.slice());
  return multiPolygon;
};
```

**Comment by Kenny806** [comment]
I have prepared a method that does a deep clone of nested arrays, but I am not sure where to place it. So far, it is in the MultiPolygon class, but I had in mind some place where it could be accessible by other classes, if needed - maybe ol.Object? 

I was also considering improving the method so that it could do a deep clone of both arrays and objects (and arrays of objects etc.). Any thoughts? 

P.S. the commit can be seen here https://github.com/Kenny806/ol3/commit/e15e914cade354e211de2369af8213c8dff963ba . I don't want to make a pull request yet, as I think the cloning method should be in a different class.

**Comment by tsauerwein** [comment]
Maybe you could use `goog.object.unsafeClone` instead?

http://docs.closure-library.googlecode.com/git/namespace_goog_object.html#goog.object.unsafeClone

**Comment by Kenny806** [comment]
Thanks, that sounds much better. I was looking for a deep clone in goog.array, didn't think of looking in goog.object (here's to reinventing the wheel). Anyway, I updated my commit to make it use goog.object.unsafeClone, see https://github.com/Kenny806/ol3/commit/06436b3217cb2abec83031ea77fd42aa98a27dcf

**Comment by Kenny806** [comment]
Created a PR: https://github.com/openlayers/ol3/pull/3359

**Comment by tschaub** [comment]
Fixed with #3359.
